poly-suite/cross-chain-event/src/agent.js

```diff
@@ -5,95 +5,122 @@ const { chains } = require("./agent-config")
+    }
+
+    findings = (await Promise.all(promises)).flat()
+    blockHeight = newBlockHeight
+    return findings
+  }
+}
+
+const createAlert = (event, storageHash, srcChainHash) => {
+  return Finding.fromObject({
+    name: "Poly Cross Chain Event With Wrong Parameters",
+    description: `The parameters of the CrossChainEvent and the poly storage are different`,
+    alertId: "POLY-CROSS-CHAIN-EVENT-WRONG-PARAMS",
+    protocol: "poly",
+    severity: FindingSeverity.Critical,
+    type: FindingType.Exploit,
+    metadata: {
+      txHash: event['TxHash'],
+      storageHash,
+      srcChainHash
+    },
+  })
+}
 
+const getBlockHeight = async (client) => {
+  const count = await client.request({ method: "getblockcount", params: [] })
+  return count - 1
+}
+
+const getEventsByBlockHeight = (block, client, contracts) => new Promise(
+  async (resolve) => {
+    const findings = []
+
+    // 'getsmartcodeevent' returns an array of events or null if there are no events
+    const events = await client.request({ method: "getsmartcodeevent", params: [block] })
+
+    // If there are events, iterate over every event[Notify]
+    if (events) {
       for (const event of events) {
+        // Every event contains an array of 'Notify' objects
         for (const notify of event["Notify"]) {
+          // Every Notify object has a 'ContractAddress' and a
+          // 'States' array = [method, srcChainId, dstChainId, txId, blockHeight, key]
           const contractAddress = notify["ContractAddress"]
           if (contractAddress !== polyCcmAddress) continue
           if (notify["States"].length < 6) continue
 
           const method = notify["States"][0]
           if (method !== "makeProof") continue
 
+          // Skip if there is no provider for the chain
           const srcChainId = notify["States"][1]
           const contract = contracts[srcChainId]
           if (!contract) continue
 
+          // Use the key to get the storage
           const key = notify["States"][5]
           const storage = await client.request({
-            method: "getstorage", 
+            method: "getstorage",
             params: [contractAddress, key.substr(40)]
           })
 
+          // The storage contains:
+          // txHash on Poly
+          // srcChainId
+          // makeTxParam (txIndex, crossChainId, fromContractAddress, toChainId, 
+          // toContractAddress, method, txData)
+          // Only hash the makeTxParam
           const rawData = storage.substr(82)
           const storageHash = ethers.utils.keccak256("0x" + rawData)
 
+          // Get the txIndex from the makeTxParam
+          // (skip the first byte, it is for size)
           const crossChainTxIndex = ethers.BigNumber.from("0x" + rawData.substr(2, 64))
           const srcChainHash = await contract.getEthTxHash(crossChainTxIndex.toNumber())
 
-          if (storageHash !== srcChainId) {
+          console.log(block, storageHash, srcChainId)
```

Comment:
Do we need to keep the console.log or was it just for development

Reply:
It was just for development